recipes/nodejs/all/conanfile.py

```diff
@@ -1,48 +1,78 @@
+        buff = re.search('GLIBC \d{1,3}.\d{1,3}', buff.getvalue())
+        return str(buff.group()[6:])
 
     def validate(self):
-        if (not (self.version in self.conan_data["sources"]) or
-            not (str(self.settings.os) in self.conan_data["sources"][self.version]) or
-            not (self._nodejs_arch in self.conan_data["sources"][self.version][str(self.settings.os)] ) ):
+        if not self.version in self.conan_data["sources"] or \
+           not str(self.info.settings.os) in self.conan_data["sources"][self.version] or \
+           not self._nodejs_arch in self.conan_data["sources"][self.version][str(self.info.settings.os)]:
             raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
 
+        if Version(self.version) >= "18.0.0":
+            if str(self.info.settings.os) == "Linux":
+                if Version(self._glibc_version) < '2.27':
+                    raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
+
     def build(self):
-        tools.get(**self.conan_data["sources"][self.version][str(self.settings.os)][self._nodejs_arch], destination=self._source_subfolder, strip_root=True)
+        pass
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version][str(self.info.settings.os)][self._nodejs_arch], destination=self._source_subfolder, strip_root=True)
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="*", src=os.path.join(self._source_subfolder, "bin"), dst="bin")
-        self.copy(pattern="node.exe", src=self._source_subfolder, dst="bin")
-        self.copy(pattern="npm", src=self._source_subfolder, dst="bin")
-        self.copy(pattern="npx", src=self._source_subfolder, dst="bin")
+        if conan_version.major == "1":
+            self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+            self.copy(pattern="*", src=os.path.join(self._source_subfolder, "bin"), dst="bin")
+            self.copy(pattern="*", src=os.path.join(self._source_subfolder, "lib"), dst="lib")
+            self.copy(pattern="node.exe", src=self._source_subfolder, dst="bin")
+            self.copy(pattern="npm", src=self._source_subfolder, dst="bin")
+            self.copy(pattern="npx", src=self._source_subfolder, dst="bin")
+        else:
+            copy(self, pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+            copy(self, pattern="*", src=os.path.join(self._source_subfolder, "bin"), dst="bin")
+            copy(self, pattern="*", src=os.path.join(self._source_subfolder, "lib"), dst="lib")
+            copy(self, pattern="node.exe", src=self._source_subfolder, dst="bin")
+            copy(self, pattern="npm", src=self._source_subfolder, dst="bin")
+            copy(self, pattern="npx", src=self._source_subfolder, dst="bin")
```

Comment:
The problem I see is not the `self.copy` vs. `copy(self` but the `dst` parameter. The actual folder must be prefixed with `self.package_folder`. See the [packaging example](https://github.com/conan-io/conan-center-index/blob/413775941483fbe74b7005c0864c6cc6fa755f3f/docs/package_templates/header_only/all/conanfile.py#L88)

```suggestion
        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self._source_subfolder)
        copy(self, pattern="*", dst=os.path.join(self.package_folder, "bin"), src=os.path.join(self._source_subfolder, "bin"))
        copy(self, pattern="*", dst=os.path.join(self.package_folder, "lib"), src=os.path.join(self._source_subfolder, "lib"))
        copy(self, pattern="node.exe", dst=os.path.join(self.package_folder, "bin"), src=self._source_subfolder)
        copy(self, pattern="npm", dst=os.path.join(self.package_folder, "bin"), src=self._source_subfolder)
        copy(self, pattern="npx", dst=os.path.join(self.package_folder, "bin"), src=self._source_subfolder)

```

Reply:
> The problem I see is not the `self.copy` vs. `copy(self` but the `dst` parameter. The destination folder must be prefixed with `self.package_folder`. See the packaging example: https://github.com/conan-io/conan-center-index/blob/413775941483fbe74b7005c0864c6cc6fa755f3f/docs/package_templates/header_only/all/conanfile.py#L88
> 
> 
> 
> ```suggestion
> 
>         copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self._source_subfolder)
> 
>         copy(self, pattern="*", dst=os.path.join(self.package_folder, "bin"), src=os.path.join(self._source_subfolder, "bin"))
> 
>         copy(self, pattern="*", dst=os.path.join(self.package_folder, "lib"), src=os.path.join(self._source_subfolder, "lib"))
> 
>         copy(self, pattern="node.exe", dst=os.path.join(self.package_folder, "bin"), src=self._source_subfolder)
> 
>         copy(self, pattern="npm", dst=os.path.join(self.package_folder, "bin"), src=self._source_subfolder)
> 
>         copy(self, pattern="npx", dst=os.path.join(self.package_folder, "bin"), src=self._source_subfolder)
> 
> 
> 
> ```

Thank you very much. Let's give it a try :)